lib/node_modules/@stdlib/repl/lib/eager_evaluator.js

```diff
@@ -0,0 +1,330 @@
+				// Iterating through arguments:
+				for ( i = 0; i < node.arguments.length; i++ ) {
+					if ( !traverse( node.arguments[ i ] ) ) {
+						return false;
+					}
+				}
+				return true;
+			}
+		}
+		return false;
+	}
+
+	/**
+	* Gets a function name representing identifier and member expression nodes.
+	*
+	* @private
+	* @param {Object} node - ast node
+	* @returns {string} - function name representing the node
+	*/
+	function getFunctionName( node ) {
+		if ( !node ) {
+			return '';
+		}
+		if ( node.type === 'MemberExpression' ) {
+			return getFunctionName(node.object) + '_' + node.property.name;
+		}
+		if ( node.type === 'Identifier' ) {
+			return node.name;
+		}
+		return '';
+	}
+} );
+
+/**
+* Clears eagerly-evaluated output.
+*
+* @private
+* @name clear
+* @memberof EagerEvaluator.prototype
+* @type {Function}
+* @returns {void}
+*
+*/
+setNonEnumerableReadOnly( EagerEvaluator.prototype, 'clear', function clear() {
+	var cursorPosition;
+
+	cursorPosition = this._rli.cursor;
+	readline.moveCursor( this._repl._ostream, 0, 1 );
+	readline.clearLine( this._repl._ostream, 0 );
+	readline.moveCursor( this._repl._ostream, 0, -1 );
+	readline.cursorTo( this._repl._ostream, cursorPosition + this._repl.promptLength() );
+	this._isPreviewing = false;
+} );
+
+/**
+* Disable the eager evaluator.
+*
+* @name disable
+* @memberof EagerEvaluator.prototype
+* @type {Function}
+* @returns {EagerEvaluator} eager evaluator instance
+*/
+setNonEnumerableReadOnly( EagerEvaluator.prototype, 'disable', function disable() {
+	this._enabled = false;
+	return this;
+});
+
+/**
+* Enables the eager evaluator.
+*
+* @name enable
+* @memberof EagerEvaluator.prototype
+* @type {Function}
+* @returns {EagerEvaluator} eager evaluator instance
+*/
+setNonEnumerableReadOnly( EagerEvaluator.prototype, 'enable', function enable() {
+	this._enabled = true;
+	return this;
+});
+
+/**
+* Callback which should be invoked **before** a "keypress" event.
+*
+* @private
+* @name beforeKeypress
+* @memberof EagerEvaluator.prototype
+* @type {Function}
+* @param {string} data - input data
+* @param {(Object|void)} key - key object
+* @returns {void}
+*
+*/
+setNonEnumerableReadOnly( EagerEvaluator.prototype, 'beforeKeypress', function beforeKeypress() {
+	if (!this._isPreviewing ) {
+		return;
+	}
+	if ( this._isPreviewing ) {
+		this.clear();
+	}
+});
+
+/**
+* Callback for handling a "keypress" event.
+*
+* @private
+* @name onKeypress
+* @memberof EagerEvaluator.prototype
+* @type {Function}
+* @param {string} data - input data
+* @param {(Object|void)} key - key object
+* @returns {void}
+*
+*/
+setNonEnumerableReadOnly( EagerEvaluator.prototype, 'onKeypress', function onKeypress() {
+	var cursorPosition;
+	var executable;
+	var nlInd;
+	var code;
+	var cmd;
+	var pre;
+	var res;
+	var tmp;
+
+	if ( !this._enabled || this._rli.line === '' ) {
+		return;
+	}
+
+	// Build the final command:
+	cmd = deepCopy( this._cmd );
```

Comment:
I believe `this._cmd` is a plain array, correct? If so, use `array/base/copy` or `array/base/slice` instead for better performance. Only use `deepCopy` is you need to actually deep copy.